pkgs/build-support/docker/stream_layered_image.py

```diff
@@ -130,6 +131,45 @@ def extract(self):
 LayerInfo = namedtuple("LayerInfo", ["size", "checksum", "path", "paths"])
 
 
+def add_base_layers(tar, from_image):
+    """
+    Adds the layers from the given base image to the final image.
+
+    tar: 'tarfile.TarFile' object for new layers to be added to.
+    from_image: Path to the base image archive.
+    """
+    if from_image is None:
+        print("No 'fromImage' provided", file=sys.stderr)
+        return []
+
+    base_tar = tarfile.open(from_image)
+
+    manifest_json_tarinfo = base_tar.getmember("manifest.json")
+    with base_tar.extractfile(manifest_json_tarinfo) as f:
+        manifest_json = json.load(f)
+
+    config_tarinfo = base_tar.getmember(manifest_json[0]["Config"])
+    with base_tar.extractfile(config_tarinfo) as f:
+        config = json.load(f)
```

Comment:
In `buildImage`, the environment in the base image is preserved. https://github.com/NixOS/nixpkgs/blob/810d22fb35287e14125470e3b917b90488ae5ed5/pkgs/build-support/docker/default.nix#L650

`fromImage` should behave the same in `buildLayeredImage`.

Reply:
Good point - added that behavior & a test for it in 389c1c0